Civi/Token/TokenProcessor.php

```diff
@@ -111,6 +111,25 @@ class TokenProcessor {
 
   protected $next = 0;
 
+  /**
+   * Get a mapping from entity_table to context (currently used by actionSchedule to map parameters)
+   *
+   * @param string $entityTable eg. civicrm_activity
+   *
+   * @return string|null
+   */
+  public static function getEntityTableToSchemaMapping($entityTable) {
+    $mapping = [
+      'civicrm_activity' => 'activityId',
+      'civicrm_contact' => 'contactId',
+      'civicrm_contribution' => 'contributionId',
+    ];
+    if (array_key_exists($entityTable, $mapping)) {
+      return $mapping[$entityTable];
+    }
+    return NULL;
+  }
```

Comment:
tldr: This seems to belong `ActionSchedule.php` rather than `TokenProcessor.php`? And it probably merits `@internal`.

long: if I understand correctly, the idea is roughly as follows: "Every actionSchedule is tied to some main entity (activity, contact, contribution, etc). Depending on what the main entity is, we need to activate some tokens (`{activity.*}`, `{contribution.*}`, etc) by passing some token-params (`activityId`, `contributionId`, etc). Put these mappings into an array/list."

To my eye, this mapping seems plausible within the `ActionSchedule` subsystem but wouldn't generalize to the `TokenProcessor` subsystem as a whole. For example, suppose we were sending an automated email about a new "Followup Activity". The message template could have tokens referencing both the original activity and the new activity. You could not use the exact same token-names/token-data for both activity records. (*They're different activities...*). It would have to be split, eg

* New activity record <=> `{activity.subject}` <=> `$tokenContext['activityId']`
* Original activity record <=> `{original_activity.subject}` <=> `$tokenContext['originalActivityId']`

For token handling generally, there isn't an intrinsic 1-1 mapping between table-names and token-data. However, for scheduled-reminders specifically, it may be a reasonable cleanup/simplification/baseline to have this mapping.

(Put another way: TokenProcessor doesn't dictate that `{$entity}Id` is part of the schema, because this would be too limiting. Rather, each token-user has loosely-coupled providers. It's up to them to decide what values are expected.)